model/riscv_addr_checks.sail

```diff
@@ -18,7 +18,7 @@ type ext_fetch_addr_error = unit
  *           any address translation errors are reported for pc, not the returned value.
  */
 function ext_fetch_check_pc(start_pc : xlenbits, pc : xlenbits) -> Ext_FetchAddr_Check(ext_fetch_addr_error) =
```

Comment:
Should the parameters here be changed to virtaddr instead? And in the other three functions below.

Reply:
Hmm, my thought is that although we know the parameters of these four functions are supposed to be virtual addresses, the naming of this function implies that it should be responsible for validating the incoming register values, which is xlenbits,  through type checks and then converting them into virtual addresses, or redirecting to other error types. Although the current implementation does not achieve this, we should keep this as an unimplemented interface for future expansion

Reply:
I don't think PC is an architecturally visible register in the same way that `x1` is though - it's kind of an invisible thing that only stores addresses (which aren't necessarily xlenbits).

I haven't made a PR for it yet but in our fork we explicitly handle that via an invalid address conversion function that is called in `set_next_pc()`.

But... it's pretty minor. I don't mind if you leave it.